src/Microsoft.ML.Tokenizers/Model/Word.cs

```diff
@@ -115,7 +115,7 @@ public void MergeAll(Dictionary<Pair<int>, (int, int)> merges, float? dropout)
             while (queue.Count > 0)
             {
                 Merge top = queue.Dequeue();
-                if (dropout.HasValue && _random.NextDouble() < dropout)
+                if (dropout.HasValue && (_random ??= new()).NextDouble() < dropout)
```

Comment:
Should this "check" be hoisted out of the loop, roughly
```c#
Random random = _random ??= new();

while (queue.Count  > 0)
{
    // use random
}
```

Reply:
I don't think it really matters. That would save a possible read/null check per iteration at the expense of doing it sometimes when it's not actually needed.

Reply:
My motivation for that question was a [reduction of code-size inside the loop](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8lAbhpuIGYnSGBhBjQDeNBuIYBtACoALWNgAmAZQzYMASzABdMRIAOUDQDd1MJuSQMAStgB2iiPgD8DAPpR7j/O2oShfhLcTCgMALLkABQAigCuMPEAPBp2GAB8DACO8fFoDI6xwAA25opQEPoQsRgAlAH+ooH+4gDusholDJHZCTAAdPxVqQwZdDV6zQyNk5MaAGZdkR5eTgzOzgC8DHYwLZE1NX0AcjAIGAAiVcUw+wyJ+eWV1eNNM+LTbzNuDFs98X3nGB/G41XyfCQAXwmzShr1h/mhwWIoTCpBiORgyVSGWBeQK1weFSqtXqEg+k1sDlWnip+B+7hp3jWm22u32YJm0P8bQ65m6GIGQwwIwYYy5ZPFswWkUZTmOpwuVxKt3uZSJz0lzXJ4Ik31+AsBwPZmv88JmZvEsIhQA).
But -- perf-wise -- there's more room for improvement, so for simplicity and easiness to read just keep it as is.

Reply:
Right, my goal here was to fix the correctness problem with as little churn as possible.